player/lua/stats.lua

```diff
@@ -1776,19 +1776,25 @@ mp.register_event("video-reconfig",
 
 --  --script-opts=stats-bindlist=[-]{yes|<TERM-WIDTH>}
 if o.bindlist ~= "no" then
-    mp.command("no-osd set really-quiet yes")
-    if o.bindlist:sub(1, 1) == "-" then
-        o.bindlist = o.bindlist:sub(2)
-        o.no_ass_b0 = ""
-        o.no_ass_b1 = ""
-    end
-    local width = max(40, math.floor(tonumber(o.bindlist) or 79))
-    mp.add_timeout(0, function()  -- wait for all other scripts to finish init
+    mp.register_event("initialized", function()
+        mp.set_property("msg-level", "all=no,statusline=status")
```

Comment:
These (the "disable all unrelated output" commands) should be set as early as possible to suppress unrelated messages as early as possible.

Originally it was set during script init, but now it's set at the event handler for `initialized` - which can be too late.

Also, please add a comment explaining what these commands do and why. It's entirely not obvious when reading the script.

Reply:
> Originally it was set during script init, but now it's set at the event handler for initialized - which can be too late. At script init doesn't guarantee it would be before everything else, but it would be as early as we can make it.

I will change it, but frankly as you said, it will race with other scripts anyway.